### ISSUE TITLE
Request body property is coerced e.g. ("123" coerced to123)

### ISSUE THREAD

**Issue by AKGP** [cdimascio/express-openapi-validator#249]
Hi,  I am using openapi: 3.0.2 with express-openAPI-validator ^2.0.0
I am getting the following problem:
When I am using 
```
new OpenApiValidator({
      apiSpec: this.openApiPath,
      validateResponses: false,
      **validateRequests:** true,
    }).install(this.app);
```

its changing the integer values to string
like 
```
body: { a: 1}
```
and I am getting 

```
body: {a:"1"}
```

Please fix this problem as I am getting validation issue due to this

**Comment by cdimascio** [comment]
@AKGP, can u share your spec. I wonder if the property `a` is specified as a string and not a number

**Comment by AKGP** [comment]
```
properties:
monthlyIncome:
          type: string
          pattern: ^[0-9]+\.?[0-9]*$
```
i am using this but when I put the value as 100 in request body I found it got changed to "100"after validation instead it show throw an error of type mismatch.

**Comment by cdimascio** [comment]
Because you spec specifies monthly income as a string. AJV will coerce the value according to its [coercion rules](https://github.com/epoberezkin/ajv/blob/master/COERCION.md). In this case, `100`can be coerced to the correct type, thus its coerced from the number `100` to expected type string, `"100"`. 

This is currently the expected behavior,

**Comment by chittolina** [comment]
@cdimascio First of all - really loved this lib. ty so much.

Now, I have a question. Why would the values be coerced *before* the validation? If that's the expected behavior, basically there's no way to use this to return errors in case the user inputs a `number` where we expect a `string`? 

I'm having the same issue right now. This is my definition:

```
 PropertySearchInput:
     type: object
     required:
         - deliveryLine
     properties:
         zip:
             type: string
             pattern: '^\d{5}$'
         deliveryLine:
             type: number
```

Using this request:

```
{"userID": "123", "deliveryLine": "445"}
```

Works fine, which, IMO, doesn't seem right. If I specify that my property is a `number`, why would a `string` be allowed?

Is it possible to add a flag to enable/disable it?

**Comment by cdimascio** [comment]
@chittolina I agree. It would be good to solve this. One possible idea is two use two separate Ajv validators for requests. One for query params and one for req bodies. This would enable coercion to continue to be enabled by default for query params (effectively a necessity) and coercion to be disabled (by default) for req bodies. Typically the body sent by a caller doesn't need to be coerced (and in your case preferable to return and error rather than coerce to the required type). Doing something like this may prove a path to solve this issue

**Comment by cdimascio** [comment]
@chittolina would you be interested in taking the first pass at solving this?

Some pointers:
1. The following [function](https://github.com/cdimascio/express-openapi-validator/blob/master/src/framework/ajv/index.ts#L7) creates an AJV validator for requests with options provided by this [caller](https://github.com/cdimascio/express-openapi-validator/blob/master/src/middlewares/openapi.request.validator.ts#L45). Basically, we need to create two validators here. 1 as-is and a second with the option `coerceTypes: false`.
2. When validation occurs e.g. [here](https://github.com/cdimascio/express-openapi-validator/blob/master/src/middlewares/openapi.request.validator.ts#L116), we need two separate validator calls:
    1. uses the existing validator and does not validate the body
    2. uses the new validator (i.e. th one with`coerceTypes:false`), and validates only the body
3. [Here](validator/blob/master/src/middlewares/openapi.request.validator.ts#L150) check both validators for errors.

**Comment by chittolina** [comment]
@cdimascio Absolutely. I'm gonna start taking a look on the contribution guideline asap and then start working on this one. I will let you know about my updates :-)

**Comment by cdimascio** [comment]
@chittolina awesome, thanks!

**Comment by cdimascio** [comment]
@chittolina are you still interested in tackling this?

**Comment by cdimascio** [comment]
PR #387 will  fix this. example [test](https://github.com/cdimascio/express-openapi-validator/blob/b909fcf4846f838b6dd7d7e51ab960b916812b23/test/coercion.spec.ts#L42)

**Comment by cdimascio** [comment]
fixed in `v4.1.0`
